typescript/sdk/src/core/AbacusCore.ts

```diff
@@ -46,15 +46,33 @@ export class AbacusCore<Chain extends ChainName = ChainName> extends AbacusApp<
     super(contractsMap, multiProvider);
   }
 
-  static fromEnvironment<Env extends CoreEnvironment>(
-    env: Env,
-    multiProvider: MultiProvider<CoreEnvironmentChain<Env>>,
-  ): AbacusCore<CoreEnvironmentChain<Env>> {
+  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+  static fromEnvironment<
+    Env extends CoreEnvironment,
+    Chain extends ChainName = ChainName,
+  >(env: Env, multiProvider: MultiProvider<Chain>) {
+    const envConfig = environments[env];
+    if (!envConfig) {
+      throw new Error(`No default env config found for ${env}`);
+    }
+
+    type EnvChain = keyof typeof envConfig;
+    type IntersectionChain = EnvChain & Chain;
+    const envChains = Object.keys(envConfig) as IntersectionChain[];
+
+    const { intersection, multiProvider: intersectionProvider } =
```

Comment:
I wonder, should this logic exist in the `AbacusApp` constructor instead? I.e.otherwise all app consumers have to implement this logic themselves?

Reply:
This is also a late thought, but I do wonder what developers would expect, i.e. do they actually expect an intersection, or would they maybe want to actually have an error thrown if config specifies a chain that the multi provider does not? I would err on the latter side tbh but curious what you and @yorhodes think.

 This also overlaps with somewhat similar logic we have in AbacusDeployer where we implicitly do the intersection as well, and it would seem useful to maybe share the logic for both call sites?